modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/DXPActivationKeysTable/index.js

```diff
@@ -0,0 +1,458 @@
+				project.accountKey,
+				licenseKeyDownloadURL,
+				encodeURI('active eq true'),
+				1,
+				9999,
+				sessionId
+			);
+			if (items) {
+				setActivationKeys({
+					data: items,
+					statusBar: {
+						activeTotalCount: items.filter((activationKey) =>
+							ACTIVATION_KEYS_LICENSE_FILTER_TYPES.active(
+								activationKey
+							)
+						).length,
+						allTotalCount: items.length,
+						expiredTotalCount: items.filter((activationKey) =>
+							ACTIVATION_KEYS_LICENSE_FILTER_TYPES.expired(
+								activationKey
+							)
+						).length,
+						notActiveTotalCount: items.filter((activationKey) =>
+							ACTIVATION_KEYS_LICENSE_FILTER_TYPES.notActivated(
+								activationKey
+							)
+						).length,
+					},
+				});
+			}
+
+			setIsLoadingActivationKeys(false);
+		};
+		fetchActivationKeysData();
+	}, [licenseKeyDownloadURL, project.accountKey, sessionId]);
+
+	useEffect(() => {
+		if (activationKeys.data.length) {
+			const activationKeysFilterData = activationKeys.data.filter(
+				(activationKey) =>
+					ACTIVATION_KEYS_LICENSE_FILTER_TYPES[filterStatusBar]
+						? ACTIVATION_KEYS_LICENSE_FILTER_TYPES[filterStatusBar](
+								activationKey
+						  )
+						: Boolean
+			);
+			setTotalCount(activationKeysFilterData.length || 5);
+			const activationKeysFilterByPage = activationKeysFilterData.slice(
+				itemsPerPage * activePage - itemsPerPage,
+				itemsPerPage * activePage
+			);
+			setActivationKeysFiltered(
+				activationKeysFilterByPage.length
+					? activationKeysFilterByPage
+					: activationKeysFilterData
+			);
+		}
+	}, [activationKeys, activePage, filterStatusBar, itemsPerPage]);
+
+	const columns = useMemo(
+		() => [
+			{
+				accessor: 'envName',
+				bodyClass: 'border-0',
+				expanded: true,
+				header: {
+					description: 'Description',
+					name: 'Environment Name',
+					styles: 'bg-transparent',
+				},
+			},
+			{
+				accessor: 'keyType',
+				bodyClass: 'border-0',
+				header: {
+					description: 'Host Name / Cluster Size',
+					name: 'Key Type',
+					noWrap: true,
+					styles: 'bg-transparent',
+				},
+			},
+			{
+				accessor: 'envType',
+				bodyClass: 'border-0',
+				header: {
+					name: 'Environment Type',
+					styles: 'bg-transparent text-neutral-10 font-weight-bold',
+				},
+			},
+			{
+				accessor: 'expirationDate',
+				bodyClass: 'border-0',
+				header: {
+					name: 'Exp. Date',
+					styles: 'bg-transparent text-neutral-10 font-weight-bold',
+				},
+				noWrap: true,
+			},
+			{
+				accessor: 'status',
+				align: 'center',
+				bodyClass: 'border-0',
+				header: {
+					name: 'Status',
+					styles: 'bg-transparent text-neutral-10 font-weight-bold',
+				},
+			},
+			{
+				accessor: 'download',
+				align: 'center',
+				bodyClass: 'border-0',
+				header: {
+					name: <ClayIcon symbol="download" />,
+					styles: 'bg-transparent text-neutral-10 font-weight-bold',
+				},
+			},
+		],
+		[]
+	);
+
+	const handleClick = async (licenseKey) => {
```

Comment:
I think generic function name like "handleClick isn't explain what it really does. Wouldn't be better a specific name like "downloadLicense" or something like that?

Reply:
Yes, I agree! I'll be changing that